pkg/awsutils/awsutils.go

```diff
@@ -1515,3 +1520,22 @@ func (cache *EC2InstanceMetadataCache) IsUnmanagedENI(eniID string) bool {
 	}
 	return false
 }
+
+func (cache *EC2InstanceMetadataCache) getENIsFromPaginatedDescribeNetworkInterfaces(
+	input *ec2.DescribeNetworkInterfacesInput) ([]*ec2.NetworkInterface, error) {
+	outputENIs := make([]*ec2.NetworkInterface, 0)
```

Comment:
micro-nit: don't create a zero capacity array, just use the nil value.  That way it will avoid an allocation, particularly in the case where you don't append any entries at all (eg: an error return).

ie:
```golang
// Don't:
foos := make([]Foo, 0)
// Do:
var foos []Foo
```

See also https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices